### PR TITLE
[dev-v2.8] update aci cni images to 6.0.3.1 for v1.27.8

### PR DESCRIPTION
**Pull request by vardhaman22**
issue: https://github.com/rancher/rancher/issues/43074

### CODE REVIEW

**Review by snasovich — APPROVED** [review]

File: pkg/rke/templates/templates.go [hunk 1]
```diff
@@ -252,7 +252,9 @@ func LoadK8sVersionedTemplates() map[string]map[string]string {
 			">=1.25.9-rancher2-2 <1.25.13-rancher1-1":  aciv5271,
 			">=1.25.13-rancher1-1 <1.26.0-rancher0":    aciv6031,
 			">=1.26.0-rancher0 <1.26.8-rancher1-1":     aciv5271,
-			">=1.26.8-rancher1-1":                      aciv6031,
+			">=1.26.8-rancher1-1 <1.27.0-rancher1-1":   aciv6031,
+			">=1.27.0-rancher1-1 <1.27.8-rancher2-1":   aciv5271,
```
Comment (manuelbuil) [inline pkg/rke/templates/templates.go:256]: this is strange. We are downgrading in 1.27.0 - 1.27.8?

File: data/data.json [hunk 2]
```diff
@@ -12647,14 +12647,12 @@
...
    "metricsServer": "rancher/mirrored-metrics-server:v0.6.3",
    "windowsPodInfraContainer": "rancher/mirrored-pause:3.7",
-   "aciCniDeployContainer": "noiro/cnideploy:5.2.7.1.81c2369",
-   "aciHostContainer": "noiro/aci-containers-host:5.2.7.1.81c2369",
-   "aciOpflexContainer": "noiro/opflex:5.2.7.1.81c2369",
-   "aciMcastContainer": "noiro/opflex:5.2.7.1.81c2369",
-   "aciOvsContainer": "noiro/openvswitch:5.2.7.1.81c2369",
-   "aciControllerContainer": "noiro/aci-containers-controller:5.2.7.1.81c2369",
-   "aciGbpServerContainer": "noiro/gbp-server:5.2.7.1.81c2369",
-   "aciOpflexServerContainer": "noiro/opflex-server:5.2.7.1.81c2369"
```
Comment (manuelbuil) [inline data/data.json:12657]: The images `aciGbpServerContainer` and `aciOpflexServerContainer` are not required anymore in `aci 6.0.3`?

**Review by kinarashah — APPROVED** [review]

File: pkg/rke/templates/templates.go (hunk 1 above)
Comment (vardhaman22) [reply pkg/rke/templates/templates.go:256]: it is a fix to update template constraint based on the system images.

File: data/data.json (hunk 2 above)
Comment (vardhaman22) [reply data/data.json:12657]: based on https://github.com/rancher/kontainer-driver-metadata/pull/1219/commits/cf48b3cb670a590028170daf5bafcf5416d7642b

**Pull request merged**